test/dynamo/test_activation_checkpointing.py

```diff
@@ -642,37 +715,56 @@ def selective_checkpointing_context_fn():
 
         def gn(x, y):
             return torch.sigmoid(
-                torch.matmul(torch.matmul(torch.bernoulli(torch.sigmoid(x)), y), y)
+                torch.matmul(
+                    torch.dropout(torch.matmul(torch.sigmoid(x), y), p=0.5, train=True),
+                    y,
+                )
             )
 
         def fn(x, y):
             return torch.utils.checkpoint.checkpoint(
                 gn,
-                torch.sin(x),
+                x,
                 y,
                 use_reentrant=False,
                 context_fn=selective_checkpointing_context_fn,
             )
 
-        x = torch.randn(4, 4, requires_grad=True)
-        y = torch.randn(4, 4, requires_grad=True)
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+        x = torch.randn(4, 4, requires_grad=True, device=device)
+        y = torch.randn(4, 4, requires_grad=True, device=device)
 
         fw_compiler = functools.partial(
             count_ops,
-            freqs=[2, 2],
-            ops=[torch.ops.aten.mm.default, torch.ops.aten.sigmoid.default],
+            freqs=[2, 2, 1],
+            ops=[
+                torch.ops.aten.mm.default,
+                torch.ops.aten.sigmoid.default,
+                torch.ops.aten.native_dropout.default,
+            ],
         )
         bw_compiler = functools.partial(
             count_ops,
-            freqs=[4, 0],
-            ops=[torch.ops.aten.mm.default, torch.ops.aten.sigmoid.default],
+            # NOTE: This unit test expects `dropout` to be recomputed (notice the count for `native_dropout` is 1).
+            # If in the future there is change to the partitioner that causes `dropout` to not be recomputed,
+            # the partitioner should expose an API for user to control what ops must be recomputed,
+            # so that we can ask the partitioner to always recompute this `dropout` op
+            # so that this unit test can still work.
+            freqs=[4, 0, 1],
+            ops=[
+                torch.ops.aten.mm.default,
+                torch.ops.aten.sigmoid.default,
+                torch.ops.aten.native_dropout.default,
+            ],
         )
         backend = aot_autograd(
             fw_compiler=fw_compiler,
             bw_compiler=bw_compiler,
             partition_fn=min_cut_rematerialization_partition,
         )
         self._validate(fn, backend, x, y)
+        if torch.cuda.is_available():
+            self._compare_orig_and_checkpointed_fns(gn, fn, x, y)
```

Comment:
Here we test that the original (non-checkpointed) version of the function gives the same output and gradients as the checkpointed version of the function.